packages/expo-background-task/android/src/main/java/expo/modules/backgroundtask/BackgroundTaskConsumer.kt

```diff
@@ -0,0 +1,116 @@
+
+  override fun taskType(): String {
+    return BACKGROUND_TASK_TYPE
+  }
+
+  /**
+   * Exposing the execute task function so that the BackgroundTaskWork (WorkManager Work unit)
+   * can execute the inner task. A task is only executed if the app is in the background.
+   * This only applies when the app is in release mode.
+   */
+  fun executeTask(callback: TaskExecutionCallback) {
+    if (isBackgrounded || ReactBuildConfig.DEBUG) {
+      Log.i(TAG, "Executing task '${task?.name}'")
+      taskManagerUtils.executeTask(task, null, callback)
+    } else {
+      Log.w(TAG, "Task was not executed since the app was not in the background or in Debug mode.")
+    }
+  }
+
+  override fun didRegister(task: TaskInterface) {
+    this.task = task
+
+    Log.i(TAG, "didRegister: ${task.name}")
+
+    // Make sure to schedule and set up the worker if it is not running
+    taskCoroutineScope.launch {
+      val context = context
+      if (!BackgroundTaskScheduler.isWorkerRunning(context)) {
+        // Get interval for the task
+        val intervalMinutes = getIntervalMinutes()
+        // Start worker
+        Log.i(TAG, "didRegister: worker not running - starting worker.")
+        BackgroundTaskScheduler.startWorker(context, task.appScopeKey, intervalMinutes)
+      } else {
+        Log.i(TAG, "didRegister: worker already running.")
+      }
+    }
+  }
+
+  override fun didUnregister() {
+    Log.i(TAG, "didUnregister: ${task?.name}")
+
+    taskCoroutineScope.launch {
+      if (!BackgroundTaskScheduler.isWorkerRunning(context)) {
+        return@launch
+      }
+
+      // Check if the task manager has more tasks like this
+      val taskService = TaskServiceProviderHelper.getTaskServiceImpl(context.applicationContext)
+        ?: throw MissingTaskServiceException()
+
+      // Get tasks for our appScope
+      val appScopeKey = task?.appScopeKey ?: throw MissingAppScopeKey()
+      task = null
+
+      // Check if we have a task that
+      val tasks = taskService.getTasksForAppScopeKey(appScopeKey)
+      val ourTasks = tasks.filter { it.getString("taskType") == BACKGROUND_TASK_TYPE }
+      if (ourTasks.isEmpty()) {
+        Log.i(TAG, "didUnregister: ${task?.name} - stopping worker, no more $BACKGROUND_TASK_TYPE tasks running.")
+        // We should just stop the worker
+        BackgroundTaskScheduler.stopWorker(context)
+      } else {
+        Log.i(TAG, "didRegister: Leaving worker running.")
+      }
+    }
+  }
+
+  override fun onHostResume() {
+    isBackgrounded = false
+  }
+
+  override fun onHostPause() {
+    isBackgrounded = true
+  }
+
+  override fun onHostDestroy() {
+    // Do nothing
+  }
+
+  private fun getIntervalMinutes(): Long {
+    val options = if (task != null) task!!.options else null
+
+    if (options != null && options.containsKey("minimumInterval")) {
+      // minimumInterval option is in minutes
+      return (options["minimumInterval"] as Number).toLong()
+    }
+    return DEFAULT_INTERVAL_MINUTES
```

Comment:
```suggestion
		return (task?.options?.["minimumInterval"] as? Number)?.toLong() 
		  ?: DEFAULT_INTERVAL_MINUTES
```

Reply:
Had to go with this one to avoid syntax errors:

```kotlin
val options = task?.options as? Map<String, Any?>
    return (options?.get("minimumInterval") as? Number)?.toLong()
      ?: DEFAULT_INTERVAL_MINUTES
```